R/resource.R

```diff
@@ -0,0 +1,187 @@
+#' Copy resource
+#'
+#' @param project study or other variable collection
+#' @param folder the folder containing the resource
+#' @param name specific resource for copy action
+#' @param new_project new location of study or other variable collection
+#' @param new_folder name of the folder in which to place the copy, defaults to
+#' folder
+#' @param new_name name of the copy, defaults to name
+#'
+#' @examples
+#' \dontrun{
+#' armadillo.copy_resource(
+#'   project = "gecko",
+#'   folder = "core_all",
+#'   name = "table1",
+#'   new_project = "gecko",
+#'   new_folder = "core_all_v2",
+#' )
+#' }
+#'
+#' @export
+armadillo.copy_resource <- # nolint
+  function(project, folder, name,
+           new_project = project,
+           new_folder = folder,
+           new_name = name) {
+    .copy_object(project,
+                folder,
+                name,
+                new_project,
+                new_folder,
+                new_name,
+                ".rds")
+  }
+
+#' Move the resource
+#'
+#' @param project a study or collection of variables
+#' @param folder the folder containing the resource to move
+#' @param name a resource to move
+#' @param new_project the project to move the resource to
+#' @param new_folder the folder to move the resource to, defaults to folder
+#' @param new_name use to rename the file, defaults to name
+#'
+#' @examples
+#' \dontrun{
+#' armadillo.move_resource(
+#'   project = "gecko",
+#'   folder = "core_all",
+#'   name = "table1",
+#'   new_project = "gecko",
+#'   new_folder = "core_all_v2",
+#' )
+#' }
+#'
+#' @export
+armadillo.move_resource <- # nolint
+  function(project, folder, name,
+           new_project = project, new_folder = folder, new_name = name) {
+    .move_object(project,
+                 folder,
+                 name,
+                 new_project,
+                 new_folder,
+                 new_name,
+                 ".rds")
+  }
+
+#' Load a resource from a project
+#'
+#' @param project study or collection variables
+#' @param folder the folder containing the resource
+#' @param name name of the resource
+#' @param env The environment in which you want to load the resource.
+#' Default is the parent.frame() from which the function is called.
+#' @return NULL, invisibly
+#'
+#' @examples
+#' \dontrun{
+#' armadillo.load_resource(
+#'   project = "gecko",
+#'   folder = "core_all",
+#'   name = "lc_core_1"
+#' )
+#'
+#' armadillo.load_resource(
+#'   project = "gecko",
+#'   folder = "core_all",
+#'   name = "lc_core_1",
+#'   env = globalenv()
+#' )
+#' }
+#'
+#' @export
+armadillo.load_resource <- function(project, folder, name, # nolint
+                                    env = parent.frame()) {
+  .load_object(project, folder, name, env, .load_resource, ".rds")
+}
+
+#' Helper function to extract an RDS file
+#'
+#' @param file file to extract
+#'
+#' @return the contents of the file
+#'
+.load_resource <- function(file) {
+  readRDS(tools::file_path_as_absolute(file))
```

Comment:
readRDS works with a connection so I think you could use the [aws.s3::s3connection](https://github.com/cloudyr/aws.s3/blob/8c722f8e84ea9f645ea394ef0908cfc003251155/R/get_object.R#L183) method instead of first writing it to harddisk

Reply:
But when you write to disk the methods are more similar cause arrow cannot do a single streaming swoop read.